ompi/mpi/c/info_get_nthkey.c

```diff
@@ -92,6 +93,11 @@ int MPI_Info_get_nthkey(MPI_Info info, int n, char *key)
 
     /* Everything seems alright. Call the back end key copy */
 
-    err = ompi_info_get_nthkey (info, n, key);
+    opal_cstring_t *key_str = NULL;
+    err = ompi_info_get_nthkey (info, n, &key_str);
+    if (NULL != key_str) {
+        opal_string_copy(key, key_str->string, MPI_MAX_INFO_KEY);
```

Comment:
Just curious -- why use `opal_string_copy()` here but not with, for example, `MPI_Info_get()`?

Reply:
Good point. `MPI_Info_get` for example used `strncpy` previously so I just kept it there. I will make it consistent to use `opal_string_copy` where relevant.

Reply:
I remember the discussion about these info objects in the MPI forum, but not all the details. In any case, from what I remember we need the original info as it will be propagated to duplicated objects, or something like this.